sources/packages/api/src/services/institution-location/institution-location.service.ts

```diff
@@ -52,4 +52,22 @@ export class InstitutionLocationService extends RecordDataModelService<Instituti
 
     return await this.repo.save(newLocation);
   }
+
+  async getAllInstitutionlocations(userInfo: UserInfo): Promise<any> {
+    //To retrive institution id
+    const institutionDetails = await this.institutionService.getInstituteByUserName(
+      userInfo.userName,
+    );
+    if (!institutionDetails) {
+      throw new Error(
+        "Not able to find a institution associated with the current user name.",
+      );
+    }
+    return this.repo
+      .createQueryBuilder("institution_location")
+      .select(['institution_location.name', 'institution_location.data', 'institution.institutionPrimaryContact'])
+      .leftJoin("institution_location.institution", "institution", "institution.id = institution_location.institution_id")
```

Comment:
I believe that TypeOrm would be guessing what the relationship looks like and we don`t need to explicitly define it. Can you check if it would be possible?
Maybe something like below would be enough:

`
.leftJoin("institution_location.institution", "institution")
`

Reply:
I agree with @andrewsignori-aot . `JoinColumn` decorator already define that relationship mapping.